core/templates/dev/head/domain/objects/NumberWithUnitsObjectFactory.js

```diff
@@ -279,12 +285,37 @@ oppia.factory('UnitsObjectFactory', [function() {
     return unit.trim();
   };
 
+  Units.createCurrencyUnits = function() {
+    // Creates user-defined currency (base + sub) units.
+    math.createUnit('dollar', {aliases: [
+      'dollars', 'Dollar', 'USD', '$', 'Dollars']});
```

Comment:
You said that special symbol such as `$` is not allowed but you are still using it as an alias. Does it work?

Reply:
In aliasing, it is allowed. But when you'll actually type `$`, it won't be allowed even though it is in alias. May be it's a bug in `math.js` that it is allowed in alias.

Reply:
Hmm, strange. What is the point of alias, then?

Reply:
Yes, `$` is not allowed. But other aliases are working well.

Reply:
Well, then, I suggest removing `$`. It leads to confusion for developers.